### PR TITLE
ENT-3256 Small performance enhancement and OS preparation for ENT changes

### PR DESCRIPTION
**Pull request by rick-r3**

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/services/statemachine/FlowLogicRefFactoryImpl.kt [hunk 1]
```diff
@@ -32,7 +34,10 @@ data class FlowLogicRefImpl internal constructor(val flowLogicClassName: String,
  * in response to a potential malicious use or buggy update to an app etc.
  */
 // TODO: Replace with a per app classloader/cordapp provider/cordapp loader - this will do for now
-class FlowLogicRefFactoryImpl(private val classloader: ClassLoader) : SingletonSerializeAsToken(), FlowLogicRefFactory {
+class FlowLogicRefFactoryImpl(private val classloader: ClassLoader, cacheFactory: NamedCacheFactory) : SingletonSerializeAsToken(), FlowLogicRefFactory {
+    init {
+        usedInEnterprise(cacheFactory)
```
Comment (rick-r3) [inline node/src/main/kotlin/net/corda/node/services/statemachine/FlowLogicRefFactoryImpl.kt]: What do we think of this?  Trying to stop having unused params that refactored away in OS.

Comment (rick-r3) [reply node/src/main/kotlin/net/corda/node/services/statemachine/FlowLogicRefFactoryImpl.kt]: Thought of a better way in this instance.

File: node/src/main/kotlin/net/corda/node/services/statemachine/FlowLogicRefFactoryImpl.kt [hunk 2]
```diff
@@ -1,8 +1,8 @@
 package net.corda.node.services.statemachine
 
-import net.corda.core.internal.VisibleForTesting
 import com.google.common.primitives.Primitives
 import net.corda.core.flows.*
+import net.corda.core.internal.VisibleForTesting
```
Comment (blsemo) [inline node/src/main/kotlin/net/corda/node/services/statemachine/FlowLogicRefFactoryImpl.kt]: Is that import used?

Comment (rick-r3) [reply node/src/main/kotlin/net/corda/node/services/statemachine/FlowLogicRefFactoryImpl.kt]: Yep.

**Comment by rick-r3** [comment]
Yes.  We want to change the implementations in Enterprise without having a copy of the code or too much merge conflict inducing differences.

**Pull request merged**